File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk c++/calcolatrice/PoligoniConvessi/quadrilatero.cpp
#include "quadrilatero.h"
Quadrilatero::Quadrilatero(const Quadrilatero & T):PoligonoConvesso(T){set_lati(T.Lista);}


Quadrilatero::Quadrilatero(const QVector<Vertice>&V,bool b):PoligonoConvesso(V,4,"Quadrilatero",b){
  set_lati(V);
}
//double Quadrilatero::get_altezza(const Vertice& V)const{}
double Quadrilatero::get_perimetro()const{
  return lati[0]+lati[1]+lati[2]+lati[3];
}
double Quadrilatero::get_area()const{
  double A,C;
  A = Vertice::get_Angolo(Lista[0],Lista[3],Lista[1]);
  C = Vertice::get_Angolo(Lista[2],Lista[1],Lista[3]);
  return ((lati[3]*lati[0])/2)*std::sin(A* (M_PI / 180.0)) + ((lati[1]*lati[2])/2)*std::sin(C* (M_PI / 180.0));

}

ostream& operator<< (ostream& os,const Quadrilatero& T){
 os<<T.toString().toUtf8().constData();
 return os;
}
QString Quadrilatero::toString() const{
  QString s ="Quadrilatero: ";
  for(auto it=Lista.begin();it!=Lista.end();it++){
    s+= it->toString();
    }
  return s;
}

void Quadrilatero::set_lati(const QVector<Vertice>& v){
  int x=0,y=1;
  while(x<v.size()){
      if(y==v.size())y=0;
      lati[x] = std::sqrt((std::pow((v[y].get_X()-v[x].get_X()),2)+std::pow(v[y].get_Y()-v[x].get_Y(),2)));
      x++; y++;
    }
}
QString Quadrilatero::get_nomeClasse()const{
    return "Quadrilatero";
}
double  Quadrilatero::get_lati(unsigned int i)const{
  return lati[i];
}
unsigned int Quadrilatero::get_nLati()const{
  return nLati;
}

File: Qt e Progetti/Progetti/2017-2018 Progetto Calcolatrice/Consegna Legrottaglie Gionata/Kalk Java/Quadrilatero.java
import java.util.Vector;
public class Quadrilatero extends PoligonoConvesso {
	protected double lati[]={0,0,0,0};
	protected int nLati=4;
	
	private void set_lati(Vector<Vertice> v) {
		  int x=0,y=1;
		  while(x<v.size()){
		      if(y==v.size())y=0;
		      lati[x] = Math.sqrt((Math.pow((v.get(y).get_X()-v.get(x).get_X()),2)+Math.pow(v.get(y).get_Y()-v.get(x).get_Y(),2)));
		      x++; y++;
		    }
	}
	public Quadrilatero(Vector<Vertice> V,boolean b) {
		super(V,4,"Quadrilatero",b);
		this.set_lati(V);
	}

	public Quadrilatero(Vector<Vertice> V) {
		super(V,4,"Quadrilatero");
	}

	Quadrilatero(Quadrilatero T){
		super(T);
		this.set_lati(T.Lista);
		}

	@Override
	public String toString() {
		  String s ="Quadrilatero: ";

		  for (Vertice i : this.Lista) {
	            
	            s+= i.toString();
		  }
		  return s;
	}

	@Override
	public String get_nomeClasse() {
		return "Quadrilatero";
	}

	@Override
	public double get_perimetro() {
		return lati[0]+lati[1]+lati[2]+lati[3];
	}

	@Override
	public double get_area() {
		  double A,C;
		  A = Vertice.get_Angolo(this.Lista.get(0),this.Lista.get(3),this.Lista.get(1));
		  C = Vertice.get_Angolo(this.Lista.get(2),this.Lista.get(1),this.Lista.get(3));
		  return ((lati[3]*lati[0])/2)* Math.sin(A* (Math.PI / 180.0)) + ((lati[1]*lati[2])/2)*Math.sin(C* (Math.PI / 180.0));
	}
	@Override
	public double get_lato(int i) {
		return this.lati[i];
	}
	@Override
	public int get_nLati() {
		return this.nLati;
	}
	@Override
	public Quadrilatero clone() {
		return new Quadrilatero(this);
	}

}
